### DISCUSSION TITLE
How to implement integrationtests for http + testcontainers + jdbi tests?

### DISCUSSION THREAD

**Discussion by bschulz-nf** [dropwizard/dropwizard#9535] [Q&A]
Hi there,

I am trying to implement an Integrationtest similar to Integrationtests in Spring Boot.
What I Would like to do is to use Testcontainers, JDBI and something lightweight to start a http server. Then I would like to do a http get. 

My code looks like this:

	@Container
	private static final MySQLContainer<?> mySQLContainer = new MySQLContainer<>(DockerImageName.parse("mysql:8.0.33"))
		.withInitScript("init-branchen.sql");

	private static ResourceExtension extension = ResourceExtension.builder()
		.setTestContainerFactory(new GrizzlyWebTestContainerFactory())
		.addResource(new BranchenRessource(new BranchenModul(branchenDao))).build();

	@RegisterExtension
	static JdbiExtension extension = JdbiTestcontainersExtension.instance(TestcontainersDatabaseInformation.of(null, "test", null, (a, b) -> "CREATE DATABASE IF NOT EXISTS test"), mySQLContainer);

	@Test
	void shouldRespondWith200() {
		// given / when
		Response found = extension.target("v3/branchen/").request().get(Response.class);

		// then
		assertEquals(200, found.getStatus());
	}

---
The BranchenRessource is a REST resource, the BranchenModul is a class containing my business logic and the BranchenDao is a jdbi interface.

Does anyone know how to achieve this?

**Comment by bschulz-nf** [comment]
@joschi do you have an idea or even examples somewhere?

**Reply by cborison** [reply]
@bschulz-nf 

I'm not entirely certain what you're looking for above; so I'm not sure this helps, but I also missed spring-boot style integration tests in dropwizard (new company) and started this project: https://github.com/dropwizard/dropwizard/discussions/9613

 - https://github.com/caspianb/dropwizard-testing

Maybe what you're asking about is in that source code (or the library itself works for your use cases?). It'd be trivial to add either a mysql extension or a more generic "testcontainers" style annotation that lets you specify the image to run, port, properties, and so on.
